packages/react/src/utils/componentUtilityFunctions.js

```diff
@@ -24,28 +24,23 @@ import dayjs from './dayjs';
  * Helper function to generate a CSV from an array of table cell data
  * Retrieve the column headers, then match and join the cell values
  * with each header
- * @param {Array<string | number>} data from table cells
+ * @param {Array<string | number | boolean>} data from table cells
  * @return {string} generated csv
  */
 export const generateCsv = (data) => {
-  let csv = '';
-  // get all headers available and merge it
-  let columnHeaders = [];
-  data.forEach((item) => {
-    columnHeaders = [...columnHeaders, ...Object.keys(item.values)];
-  });
-  columnHeaders = [...new Set(columnHeaders)];
-  csv += `${columnHeaders.join(',')}\n`;
-  data.forEach((item) => {
-    columnHeaders.forEach((arrayHeader) => {
-      // if item is of arrayHeader, add value to csv
-      // isNil will also correct the cases in which the value is 0 or false
-      csv += `${!isNil(item.values[arrayHeader]) ? item.values[arrayHeader] : ''},`;
-    });
-    csv += `\n`;
-  });
-
-  return csv;
+  const allHeadersObj = Object.assign({}, ...data.map((row) => row.values));
+  const columnHeaders = Object.keys(allHeadersObj);
+  const headerRow = `${columnHeaders.join(',')}\n`;
+  const lastColumnId = columnHeaders.slice(-1)[0];
+
+  return data.reduce((previousCsv, row) => {
+    return columnHeaders.reduce((previousNestedCsv, headerId) => {
+      const separator = headerId === lastColumnId ? '\n' : ',';
+      return `${previousNestedCsv}${row.values[headerId] ?? ''}${separator}`;
+    }, previousCsv);
+  }, headerRow);
+
+  // return csv.replace(/,$/, '\n');
```

Comment:
Let's remove the comment. Otherwise it looks great.  Good catch!
```suggestion

```

Reply:
Committed, thanks